connectors/source.py

```diff
@@ -105,20 +106,27 @@ def value(self):
     def value(self, value):
         self._value = value
 
-    def _convert(self, value, field_type):
-        if not isinstance(value, str):
-            # we won't convert the value if it's not a str
+    def _convert(self, value, field_type_):
+        cast_type = locate(field_type_)
+        if cast_type not in [str, int, float, bool, list]:
+            # unsupported type
             return value
 
-        if field_type == "int":
-            return int(value)
-        elif field_type == "float":
-            return float(value)
-        elif field_type == "bool":
-            return value.lower() in ("y", "yes", "true", "1")
-        elif field_type == "list":
-            return [item.strip() for item in value.split(",")]
-        return value
+        if isinstance(value, cast_type):  # pyright:ignore
+            return value
+
+        # list requires special type casting
+        if field_type_ == "list":
```

Comment:
nit, you can use
```suggestion
        if cast_type is list:
```

Reply:
This isn't checking if the value is a list, but if the value is supposed to be a list. Let's say the user has created a field like this:

```
{
  "label": "foo",
  "value": "bar",
  "type": "list"
}
```

In this case, the `cast_type` value will be `str` which doesn't align with the `field_type_` which would be `list`.
If the value and type align then it will already be returned on line 116.

Reply:
AH I was getting confused, I thought for a moment that `cast_type` was built from `value` rather than `field_type_`. I just woke up sorry.

Reply:
Fixed!